snap/snapcraft.yaml

```diff
@@ -117,6 +117,11 @@ parts:
               ${CRAFT_PART_INSTALL}/share/dbus-1/system-services/org.freedesktop.fwupd.service
       cp -R /usr/libexec/fwupd/efi ${CRAFT_PART_INSTALL}/libexec/fwupd
       cp /usr/lib/shim/shimx64.efi.signed ${CRAFT_PART_INSTALL}/libexec/fwupd/efi/
+
+      objcopy -O binary -j .sbatlevel /usr/lib/shim/shimx64.efi.signed sbatlevel.section
+      python3 "${CRAFT_PROJECT_DIR}/snap/local/get_sbat_var_previous.py" sbatlevel.section >shimx64.efi.signed.sbatlevel
+      cp shimx64.efi.signed.sbatlevel /usr/lib/shim/shimx64.efi.signed ${CRAFT_PART_INSTALL}/libexec/fwupd/efi/
```

Comment:
this seems kinda fragile. Could the fwupd code use libelf to get the specific section data?

Reply:
If it is ok to add dependencies, yes.

Reply:
Maybe; ELF isn't super tricky to parse. Give me a few hours and I'll see what I can do.

Reply:
I can do it with just the linux headers (`linux/elf.h`). It is not difficult. It is just a bit annoying.

Reply:
Hang on -- isn't the fwupdx64.efi going to be in PE format? we already have a parser for that.

Reply:
I forgot, it is a PE/COFF....

Reply:
So probably with `linux/coff.h`.

Reply:
Nice. I will look at that.

Reply:
Nah! You can do `fu_pefile_firmware_new()`, then `fu_firmware_parse()`, then `fu_firmware_get_image_by_id_bytes(.sbat)`

Reply:
In case you did need ELF for some strange reason, I just rather enjoyed the last hour :) https://github.com/fwupd/fwupd/pull/5959

Reply:
Done. But I had to fix the long names. It is `.sbatlevel` section we need to read. not `.sbat`.